### PR TITLE
Options for testing effects of analysis and solved examples

### PR DESCRIPTION
**Pull request by jmontleon**
We want to be able to test the value of including different components of the prompt, such as analysis data, and solved examples.

Options such as these can be used to limit what is included. Leaving it blank will include all components.

```
provider = "IBMOpenSource"
args = { model_id = "ibm-mistralai/mixtral-8x7b-instruct-v01-q", template = "preamble_only" }
```

```
provider = "IBMOpenSource"
args = { model_id = "ibm-mistralai/mixtral-8x7b-instruct-v01-q", template = "preamble_with_analysis_only" }
```

### CODE REVIEW

File: kai/data/templates/preamble.txt [hunk 1]
```diff
@@ -1,17 +1,21 @@
-# Java EE to Quarkus Migration
-You are an AI Assistant trained on migrating enterprise JavaEE code to Quarkus. I will give you an example of a JavaEE file and you will give me the Quarkus equivalent.
+2
```
Comment (JonahSussman) [inline kai/data/templates/preamble.txt:1]: Stray "2"?

File: kai/server.py [hunk 2]
```diff
@@ -497,15 +505,38 @@ async def get_incident_solutions_for_file(request: Request):
...
+                prompt = build_prompt(CONFIG_IBM_LLAMA_MF, args)
+        else:
+            if config_template == "preamble_only":
+                prompt = build_prompt(CONFIG_IBM_GRANITE_MF_PREAMBLE_ONLY, args)
+            elif config_template == "preamble_with_analysis_only":
+                prompt = build_prompt(
+                    CONFIG_IBM_GRANITE_MF_PREAMBLE_WITH_ANALYSIS_ONLY, args
+                )
+            else:
+                prompt = build_prompt(CONFIG_IBM_GRANITE_MF, args)
```
Comment (JonahSussman) [inline kai/server.py:539]: Might be better to put this in the model provider classes. Then we can just call `build_prompt(provider.get_model_config(), args)`

File: kai/server.py [hunk 3]
```diff
@@ -497,15 +505,38 @@ async def get_incident_solutions_for_file(request: Request):
...
-                "src_file_contents": updated_file,
-                "incidents": incidents,
-            },
-        )
+        config_template = ""
+        base_path = os.path.dirname(__file__)
+        with open(os.path.join(base_path, "config.toml"), "rb") as f:
+            config = tomllib.load(f)
+        if "template" in config["models"]["args"]:
+            config_template = config["models"]["args"]["template"]
```
Comment (JonahSussman) [inline kai/server.py:513]: Might be better to put this in the model provider class

File: kai/data/templates/preamble.txt (hunk 1 above)
Comment (jmontleon) [reply kai/data/templates/preamble.txt:1]: yes, thank you.

File: kai/server.py (hunk 2 above)
Comment (jmontleon) [reply kai/server.py]: Thank you for the patch @JonahSussman !

**Review by fabianvf — APPROVED** [review]

**Review by JonahSussman — APPROVED** [review]
LGTM

**Pull request merged**